tests/clients/test_ngsi_v2_iota.py

```diff
@@ -430,11 +434,30 @@ def test_update_service_group(self):
         self.client.update_group(service_group=group_base)
         self.assertEqual(group_base, self.client.get_group(resource="/iot/json", apikey="base"))
 
+    def test_clear_iot_agent(self):
+        """
+        Test for clearing iot agent AFTER clearing context broker
+        while having a device with a command
+
+        Returns:
+            None
+        """
+        device = Device(**self.device)
+        device.add_command(DeviceCommand(name="dummy_cmd"))
+        self.client.post_device(device=device)
+        clear_context_broker(settings.CB_URL,self.fiware_header)
+        self.assertRaises(requests.HTTPError,clear_iot_agent,
+                          settings.IOTA_URL,self.fiware_header)
+
     def tearDown(self) -> None:
         """
         Cleanup test server
+
         """
         self.client.close()
-        clear_all(fiware_header=self.fiware_header,
-                  cb_url=settings.CB_URL,
-                  iota_url=settings.IOTA_JSON_URL)
+        try:
+            clear_all(fiware_header=self.fiware_header,
+                    cb_url=settings.CB_URL,
+                    iota_url=settings.IOTA_JSON_URL)
+        except requests.HTTPError:
```

Comment:
Why must this `except` be added? I think in `clear_all` the oder of the clear functions is correct. And here by capturing all HTTPError, it will show "Already cleared" even if the server is not reachable. Could you refine the check for exception? Otherwise, I would suggest to roll it back

Reply:
@SystemsPurge OK, I understand now. Thanks for explaining. I actually found out that if the registration is deleted, the device can never be deleted via the API ... To make it stable, I would suggest that we can add a flag in clear_context_broker to control whether to clear registrations. By default, we can set it to false, i.e. not to clear regestrations. What do you think?

Reply:
@djs0109 That's convenient, should solve the order issue completely since apparently it only fails to delete a device ( after deleting registration ) if the device has a command.

Reply:
Yes, sounds great :)